src/components/ContentEditor/WidgetEditor/WidgetEditor.tsx

```diff
@@ -64,20 +60,36 @@ export const WidgetEditor: React.FC<WidgetEditorProps> = ({ content, contentHash
 
         {editing && <>
             <div>
-                <FormControl className="content-editor-formcontrol">
-                    <InputLabel>Select a component type</InputLabel>
-                    <Select
-                        value={editedContent.type}
-                        onChange={(e) => {
-                            setEditedContentOnChange("type", e.target.value as string, editedContent, setEditedContent);
-                        }}>
-                        {Object.keys(WidgetTypes).map(widgetType => {
-                            return <MenuItem key={widgetType} value={widgetType}>
-                                {widgetType}
-                            </MenuItem>
-                        })}
-                    </Select>
-                </FormControl>
+                <div className="wiki-form">
+                    <form>
+                        <fieldset>
+                            <legend>Select a Widget</legend>
+                            <select id="widget" className="widget-type"
+                                value={editedContent.type}
+                                onChange={(e) => {
+                                    setEditedContentOnChange("type", e.target.value as string, editedContent, setEditedContent);
+                                }}>
+                                <option value="" disabled selected>-- Select a Widget --</option>
+                                {/* display Widget Categories and Widgets in dropdown, sorted alphabetically */}
+                                {Object.keys(WidgetCategories).map((category) => {
```

Comment:
You are mapping the widget categories (O(n)), then filtering (O(n))+ sorting (O(n)) +  mapping again (O(n)) inside of that. so this is roughly a O(3n^2) operation. Is there a way we can shorten this to only have it map stuff once, or at least not do any calculations multiple times for each widget? If it loads it's fine, but this should be put on a backlog as a high priority thing to optimize.